node/subsystem-util/src/rolling_session_window.rs

```diff
@@ -370,22 +529,58 @@ async fn load_all_sessions(
 			))
 			.await;
 
-		let session_info = match rx.await {
-			Ok(Ok(Some(s))) => s,
-			Ok(Ok(None)) => return Err(SessionsUnavailableReason::Missing(i)),
-			Ok(Err(e)) => return Err(SessionsUnavailableReason::RuntimeApi(e)),
-			Err(canceled) => return Err(SessionsUnavailableReason::RuntimeApiUnavailable(canceled)),
+		match rx.await {
+			Ok(Ok(Some(session_info))) => {
+				// We do not allow failure anymore after having at least 1 session in window.
+				allow_failure = false;
+				sessions.push(session_info);
+			},
+			Ok(Ok(None)) if !allow_failure => return Err(SessionsUnavailableReason::Missing(i)),
+			Ok(Ok(None)) => {
+				// Handle `allow_failure` true.
+				// If we didn't get the session, we advance window start.
+				*window_start += 1;
+				gum::debug!(
+					target: LOG_TARGET,
+					session = ?i,
+					"Session info missing from runtime."
+				);
+			},
+			Ok(Err(e)) if !allow_failure => return Err(SessionsUnavailableReason::RuntimeApi(e)),
+			Err(canceled) if !allow_failure =>
+				return Err(SessionsUnavailableReason::RuntimeApiUnavailable(canceled)),
+			Ok(Err(err)) => {
+				// Handle `allow_failure` true.
+				// If we didn't get the session, we advance window start.
```

Comment:
This matches only if `allow_failure` is true. That flag stays true only for a continous window of missing sessions at the beginning of the rolling session window. Since such failures are not transient, we advance the window to something that.